cashews/wrapper.py

```diff
@@ -163,16 +164,25 @@ def get(self, key: str, default: Optional[Any] = None) -> Any:
     def get_raw(self, key: str) -> Any:
         return self._with_middlewares("get_raw", key)(key=key)
 
-    async def keys_match(self, pattern: str):
+    async def keys_match(self, pattern: str, *, decode: bool = False) -> AsyncGenerator[str | bytes, None]:
+        """
+        :param decode: convert keys from bytes to strings to be consistent with "Memory" and
+            "Diskcache" backends because "Redis" `SCAN` returns keys as bytes by default
+        """
         backend, middlewares = self._get_backend_and_config(pattern)
 
         async def call(_pattern):
             return backend.keys_match(_pattern)
 
         for middleware in middlewares:
             call = partial(middleware, call, cmd="keys_match", backend=backend)
-        async for key in (await call(pattern)):
-            yield key
+
+        if decode and isinstance(backend, Redis):
+            async for key in (await call(pattern)):
+                yield str(key)
```

Comment:
Actually, `str(key)` doesn't work. You have to do `key.decode()`.

Reply:
Oh, my bad. Thank you.